component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java

```diff
@@ -915,4 +915,269 @@ public void receive(Event[] events) {
+                "define stream FooStream (symbol string, price float, volume long); " +
+                "define stream BarStream (symbol string, price float, volume long); ";
+        String query = "" +
+                "from FooStream " +
+                "select * " +
+                "insert into BarStream; ";
+        SiddhiManager siddhiManager = new SiddhiManager();
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(streams + query);
+        siddhiAppRuntime.addCallback("BarStream", new StreamCallback() {
+
+            @Override
+            public void receive(Event[] events) {
+                EventPrinter.print(events);
+                for (Event event : events) {
+                    switch (count.incrementAndGet()) {
+                    case 1:
+                        AssertJUnit.assertEquals(55.6f, event.getData(1));
+                        break;
+                    case 2:
+                        AssertJUnit.assertEquals(32.6f, event.getData(1));
+                        break;
+                    default:
+                        AssertJUnit.fail();
+                    }
+                }
+            }
+        });
+        siddhiAppRuntime.start();
+        InMemoryBroker.publish("stock", "[{\n"
+                + "  \"event\":{\n"
+                + "      \"symbol\":\"WSO2\",\n"
+                + "      \"price\":55.6,\n"
+                + "      \"volume\":100}},"
+                + "{\n"
+                +   "\"event\":{\n"
+                +       "\"symbol\":\"IBM\",\n"
+                +       "\"price\":32.6,\n"
+                +       "\"volume\":160\n"
+                +   "}\n"
+                + "}]");
+
+        SiddhiTestHelper.waitForEvents(waitTime, 2, count, timeout);
+        //assert event count
+        AssertJUnit.assertEquals("Number of events", 2, count.get());
+        siddhiAppRuntime.shutdown();
+    }
+
+    @Test
+    public void jsonSourceMapperTest15() throws InterruptedException {
+        log.info("test JsonSourceMapper with test json missing attribute");
+        String streams = "" +
+                "@App:name('TestSiddhiApp')" +
+                "@source(type='inMemory', topic='stock', @map(type='json')) " +
+                "define stream FooStream (symbol string, price float, volume long); " +
+                "define stream BarStream (symbol string, price float, volume long); ";
+        String query = "" +
+                "from FooStream " +
+                "select * " +
+                "insert into BarStream; ";
+        SiddhiManager siddhiManager = new SiddhiManager();
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(streams + query);
+        siddhiAppRuntime.start();
+        InMemoryBroker.publish("stock", "12");
+
+        siddhiAppRuntime.shutdown();
+    }
+
+    @Test
+    public void jsonSourceMapperTest16() throws InterruptedException {
```

Comment:
Where is the assertion? If you do not get an exception try to assert log content. Else no point in having the test